examples/getting_started/dynamic_config_and_chaining.ipynb

```diff
@@ -0,0 +1,734 @@
+    "@prompt_template(\"Rewrite this text to be more positive: {text}\")\n",
+    "def make_positive(text: str): ...\n",
+    "\n",
+    "\n",
+    "@openai.call(\"gpt-4o-mini\")\n",
+    "@prompt_template(\"Summarize this text: {text}\")\n",
+    "def summarize(text: str): ...\n",
+    "\n",
+    "\n",
+    "def process_text(text: str):\n",
+    "    sentiment = analyze_sentiment(text).sentiment\n",
+    "    if sentiment == \"negative\":\n",
+    "        return make_positive(text).content\n",
+    "    else:\n",
+    "        return summarize(text).content\n",
+    "\n",
+    "\n",
+    "negative_text = \"I had a terrible day. Everything went wrong.\"\n",
+    "positive_text = \"Today was amazing! I accomplished all my goals.\"\n",
+    "\n",
+    "print(\"Processed negative text:\", process_text(negative_text))\n",
+    "print(\"\\nProcessed positive text:\", process_text(positive_text))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Parallel Chaining\n",
+    "\n",
+    "We can execute multiple chains in parallel for improved efficiency."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-09-10T04:44:54.113507Z",
+     "start_time": "2024-09-10T04:44:53.511780Z"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "French: Bonjour, le monde !\n",
+      "Spanish: ¡Hola, mundo!\n",
+      "German: Hallo, Welt!\n",
+      "Italian: Ciao, mondo!\n"
+     ]
+    }
+   ],
+   "source": [
+    "import asyncio\n",
+    "\n",
+    "\n",
+    "@openai.call(\"gpt-4o-mini\")\n",
+    "@prompt_template(\"Translate this text to {language}: {text}\")\n",
+    "async def translate_async(text: str, language: str): ...\n",
+    "\n",
+    "\n",
+    "async def translate_to_multiple_languages(text: str, languages: list[str]):\n",
+    "    tasks = [translate_async(text, lang) for lang in languages]\n",
+    "    results = await asyncio.gather(*tasks)\n",
+    "    return {\n",
+    "        lang: result.content for lang, result in zip(languages, results, strict=False)\n",
+    "    }\n",
+    "\n",
+    "\n",
+    "text = \"Hello, world!\"\n",
+    "languages = [\"French\", \"Spanish\", \"German\", \"Italian\"]\n",
+    "\n",
+    "\n",
+    "async def main():\n",
+    "    translations = await translate_to_multiple_languages(text, languages)\n",
+    "    for lang, translation in translations.items():\n",
+    "        print(f\"{lang}: {translation}\")\n",
+    "\n",
+    "\n",
+    "# Run the asynchronous function in Jupyter notebooks\n",
+    "await main()\n",
+    "\n",
+    "# Run the asynchronous function in a Python script\n",
+    "# asyncio.run(main())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Advanced Techniques\n",
+    "\n",
+    "### Combining Dynamic Config and Chaining\n",
```

Comment:
you already showed this above with chaining using computed fields?

I would maybe make a more clear distinction here between chaining using functions vs. chaining with computed fields, specifically focusing on the fact that when using computed fields you get the full chain as part of the response of a single function rather than having to call and track them separately in sequence